Bottom: 83185cf4ed2d2eca47586b807d379b74f4ce8747
Top:    3708ae823aab3860cce664bbc054f82a4625d743
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-05-10 12:19:58 -0500

Move sshd config to setup-remote-user role


---
diff --git a/roles/docker-common/tasks/main.yml b/roles/docker-common/tasks/main.yml
index 05894452..da2d022b 100644
--- a/roles/docker-common/tasks/main.yml
+++ b/roles/docker-common/tasks/main.yml
@@ -5,7 +5,6 @@
   when: ansible_os_family == "RedHat"
 
 - import_tasks: packages.yml
-- import_tasks: sshd_settings.yml
 
 - import_tasks: configure.yml
 
diff --git a/roles/docker-common/tasks/sshd_settings.yml b/roles/docker-common/tasks/sshd_settings.yml
deleted file mode 100644
index 933c7f23..00000000
--- a/roles/docker-common/tasks/sshd_settings.yml
+++ /dev/null
@@ -1,20 +0,0 @@
----
-# file: roles/docker-common/tasks/sshd_settings.yaml
-
-- name: Set ssh ListenAddress
-  replace:
-    dest: /etc/ssh/sshd_config
-    regexp: '#ListenAddress 0.0.0.0'
-    replace: 'ListenAddress {{ ansible_br0["ipv4"]["address"] }}'
-    backup: yes
-  when: 
-    - ansible_os_family == "RedHat"
-    - ansible_br0 is defined and ansible_br0.ipv4 is defined and ansible_br0.ipv4.address is defined
-
-- name: Restart sshd server
-  service:
-    name: sshd
-    state: restarted
-  when: 
-    - ansible_os_family == "RedHat"
-    - ansible_br0 is defined and ansible_br0.ipv4 is defined and ansible_br0.ipv4.address is defined
diff --git a/roles/setup-remote-user/tasks/main.yml b/roles/setup-remote-user/tasks/main.yml
index 11e5abd2..ed59a082 100644
--- a/roles/setup-remote-user/tasks/main.yml
+++ b/roles/setup-remote-user/tasks/main.yml
@@ -40,6 +40,15 @@
     line: 'PermitRootLogin no'
     state: present
 
+- name: Set ssh ListenAddress
+  lineinfile:
+    dest: /etc/ssh/sshd_config
+    regexp: '^[#]ListenAddress .*$'
+    replace: 'ListenAddress {{ ansible_br0["ipv4"]["address"] }}'
+  when: 
+    - ansible_os_family == "RedHat"
+    - ansible_br0 is defined and ansible_br0.ipv4 is defined and ansible_br0.ipv4.address is defined
+
 - name: restart sshd service
   service:
     name: sshd
